shell/src/main/java/alluxio/cli/fs/command/LsCommand.java

```diff
@@ -238,7 +240,19 @@ private void ls(AlluxioURI path, boolean recursive, boolean forceLoadMetadata, b
       optionsBuilder.setLoadMetadataType(LoadMetadataPType.ALWAYS);
     }
     optionsBuilder.setRecursive(recursive);
+
+    // If list status takes too long, print the message
+    Timer timer = new Timer();
+    timer.schedule(new TimerTask() {
+      @Override
+      public void run() {
+        System.out.println("Getting file information from Alluxio master, "
+            + "this may take a while....");
```

Comment:
how about "Getting directory status may take a while if it has millions of files or sub-directories."